improver_tests/utilities/test_neighbourhood_tools.py

```diff
@@ -87,3 +97,108 @@ def test_rolling_window_writable(array_size_5):
     windows = rolling_window(array_size_5, (2, 2), writeable=True)
     windows[0, 0, 0, 0] = -1
     assert windows[0, 0, 0, 0] == -1
+
+
+def test_padding_neighbourhood_size_2(array_size_5):
+    """Test that result is same as result of rolling_window with a border of zeros."""
+    padded = pad_and_roll(array_size_5, (2, 2), mode="constant")
+    window = rolling_window(array_size_5, (2, 2))
+    inner_part = padded[1:-1, 1:-1, ::]
+    np.testing.assert_array_equal(inner_part, window)
+    border_index = (
+        [[0, i, 0, j] for i in range(5) for j in [0, 1]]
+        + [[5, i, 1, j] for i in range(5) for j in [0, 1]]
+        + [[i, 0, j, 0] for i in range(5) for j in [0, 1]]
+        + [[i, 5, j, 1] for i in range(5) for j in [0, 1]]
+    )
+    outer_part = padded[list(zip(*border_index))]
+    np.testing.assert_array_equal(outer_part, np.zeros(40, dtype=np.int32))
+
+
+def test_padding_non_zero(array_size_5):
+    """Test padding with a number other than the default of 0."""
+    padded = pad_and_roll(array_size_5, (2, 2), mode="constant", constant_values=1)
+    border_index = (
+        [[0, i, 0, j] for i in range(5) for j in [0, 1]]
+        + [[5, i, 1, j] for i in range(5) for j in [0, 1]]
+        + [[i, 0, j, 0] for i in range(5) for j in [0, 1]]
+        + [[i, 5, j, 1] for i in range(5) for j in [0, 1]]
+    )
+    outer_part = padded[list(zip(*border_index))]
+    np.testing.assert_array_equal(outer_part, np.ones(40, dtype=np.int32))
+
+
+def test_pad_boxsum(array_size_3):
+    """Test that padded array consists of input array surrounded by border of zeros."""
+    padded = pad_boxsum(array_size_3, 3, mode="constant")
```

Comment:
I've noticed the `pad_boxsum` generates an array that isn't evenly padded i.e. due to `[(ih + 1, ih), (jh + 1, jh)]`. I think that it might be worth adding a few more words to the current docstring i.e. "Pad an array to shape suitable for `boxsum`." to mention that more padding is added to the top and left of the array. Perhaps something like: "Pad an array to shape suitable for `boxsum`. The resulting array is unevenly padded with an extra row/column of padding at the top/left." What do you think?